### DISCUSSION TITLE
Draft documents don't get saved to db when using the autosave option

### DISCUSSION THREAD

**Discussion by ollebergkvist** [payloadcms/payload#1031] [Q&A]
# Bug Report

When using the autosave option for drafts, field data are not saved to the db.
If you however use the manual save config option (i.e drafts: true), which generates a "save draft" button that the
user manually has to click, data is saved to the db as expected.

## Expected Behavior

Draft document fields should be saved to db when autosave is enabled. 

## Current Behavior

Field values are not saved to the db, empty fields.

https://jmp.sh/b6U7Wz8

## Steps to Reproduce

1. Add this collection to payload and import to config: https://github.com/payloadcms/payload/blob/master/test/versions/collections/Autosave.ts
2. Try to create a doc to this collection via the UI
3. Wait for the autosave function to trigger
4. Check the doc in your db manager
5. Fields will be empty

## Detailed Description

System:
    OS: macOS 12.5.1
    CPU: (6) x64 Intel(R) Core(TM) i5-8500B CPU @ 3.00GHz
    Memory: 163.45 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 14.18.3 - ~/.asdf/installs/nodejs/14.18.3/bin/node
    Yarn: 1.22.10 - /usr/local/bin/yarn
    npm: 6.14.15 - ~/.asdf/plugins/nodejs/shims/npm
  Browsers:
    Chrome: 104.0.5112.101
    Firefox: 103.0.2
    Safari: 15.6.1
  Packages:
    Payload: ^1.0.27

**Comment by jmikrut** [comment]
Hey @ollebergkvist — this is actually working as intended. With Autosave, as soon as you click "create new", a blank document is created because we need the ID to point autosave versions against. As you work, an autosave version will be saved to the `_collectionSlug_versions` collection, and that will be kept updated. It is only when you publish your changes that the autosaved changes will appear in the main `collectionSlug` collection. 

With Autosave enabled, there is no way for Payload to know when you are ready to save changes to your "main" document in the main collection. Rather, we just create a blank one, and then store all autosaved changes to the `_versions` collection.

Does this make sense? I'm going to bump this one to a Discussion so we can keep the conversation going. There could be ways to improve this, but at first pass, this is operating as intended. When you click `publish`, your changes will be copied from the `_versions` collection straight to the parent document as intended.

Thoughts?
